bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/dial/DialDiscoveryParticipant.java

```diff
@@ -0,0 +1,157 @@
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * This implementation of the {@link UpnpDiscoveryParticipant} provides discovery of Sony DIAL protocol devices.
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+@NonNullByDefault
+@Component(immediate = true, configurationPid = "discovery.sony-dial")
+public class DialDiscoveryParticipant extends AbstractDiscoveryParticipant implements UpnpDiscoveryParticipant {
+    /**
+     * Creates the discovery participant
+     */
+    public DialDiscoveryParticipant() {
+        super(SonyBindingConstants.DIAL_THING_TYPE_PREFIX);
+    }
+
+    @Override
+    protected boolean getDiscoveryEnableDefault() {
+        return false;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(final RemoteDevice device) {
+        Objects.requireNonNull(device, "device cannot be null");
+
+        if (!isDiscoveryEnabled()) {
+            return null;
+        }
+
+        final ThingUID uid = getThingUID(device);
+        if (uid == null) {
+            return null;
+        }
+
+        final RemoteDeviceIdentity identity = device.getIdentity();
+        final URL dialURL = identity.getDescriptorURL();
+
+        String deviceId;
+        try {
+            final DialClient dialClient = new DialClientFactory().get(dialURL.toString());
+            if (dialClient == null || !dialClient.hasDialService()) {
+                logger.debug(
+                        "DIAL device couldn't be created or didn't implement any device information - ignoring: {}",
+                        identity);
+                return null;
+            }
+
+            deviceId = dialClient.getFirstDeviceId();
+        } catch (final IOException e) {
+            logger.debug("DIAL device exception {}: {}", device.getIdentity(), e.getMessage(), e);
+            return null;
+        }
+
+        final DialConfig config = new DialConfig();
+
+        String macAddress = getMacAddress(identity, uid);
+        if (macAddress == null && NetUtil.isMacAddress(deviceId)) {
+            macAddress = deviceId;
+        }
+        config.setDiscoveredMacAddress(macAddress);
+        config.setDeviceAddress(dialURL.toString());
+
+        final String thingId = UidUtils.getThingId(identity.getUdn());
+        return DiscoveryResultBuilder.create(uid).withProperties(config.asProperties())
+                .withProperty("DialUDN", StringUtils.defaultIfEmpty(thingId, uid.getId()))
+                .withRepresentationProperty("DialUDN").withLabel(getLabel(device, "DIAL")).build();
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(final RemoteDevice device) {
+        Objects.requireNonNull(device, "device cannot be null");
+
+        if (!isDiscoveryEnabled()) {
+            return null;
+        }
+
+        if (isSonyDevice(device)) {
+            // if (isScalarThingType(device)) {
```

Comment:
dead code or todo?

Reply:
hmm - this is actually an interesting one.  I'm going to eliminate it (and in the other discoveries) because a recent enhancement (allowing the discoveries to be toggled on/off via a runtime setting) would make this kinda obsolete (this was a prior attempt to prefer only showing scalar service - but with the settings, the user can no decide that).